base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -230,25 +243,29 @@ impl CipherSeed {
     }
 
     fn apply_stream_cipher(data: &mut Vec<u8>, passphrase: &str, salt: &[u8]) -> Result<(), KeyManagerError> {
-        let argon2 = Argon2::default();
         let blake2_nonce_hasher: VarBlake2b = VarBlake2b::new(size_of::<Nonce>())
             .expect("Should be able to create blake2 hasher; will only panic if output size is 0 or greater than 64");
 
         let mut encryption_nonce = [0u8; size_of::<Nonce>()];
+
+        // TODO: use hashing API to generate suitable domain separation instead
+        let label = b"chacha20_salt";
```

Comment:
Is this domain separation label too generic? Other separators elsewhere are more detailed to avoid any accidental reuse elsewhere.

Reply:
Definitely agree. I changed this to, following the current convention, `com.tari.key_manager.cipher_seed.chacha20_enc`. Similarly for the MAC generation label.